### PR TITLE
migrate Strip Hit Efficiency HitEff class to use esconsumes

### PR DESCRIPTION
**Pull request by mmusich**
#### PR description:

Title says it all. Part of https://github.com/cms-sw/cmssw/issues/31061

#### PR validation:

Run the unit tests in `CalibTracker/SiStripCommon`

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

Not a backport, no backport needed.

### CODE REVIEW

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by mmusich** [comment]
@makortel, this should solve:

> I found errors in the following unit tests:
>
>---> test testCalibTrackerSiStripCommon had ERRORS

in https://github.com/cms-sw/cmssw/pull/31746#issuecomment-706615068

**Comment by makortel** [comment]
Thanks @mmusich!

File: CalibTracker/SiStripHitEfficiency/src/HitEff.cc [hunk 1]
```diff
@@ -81,6 +79,16 @@ HitEff::HitEff(const edm::ParameterSet& conf)
           consumes<edmNew::DetSetVector<SiStripCluster> >(conf.getParameter<edm::InputTag>("siStripClusters"))),
       digis_token_(consumes<DetIdCollection>(conf.getParameter<edm::InputTag>("siStripDigis"))),
       trackerEvent_token_(consumes<MeasurementTrackerEvent>(conf.getParameter<edm::InputTag>("trackerEvent"))),
+      topoToken_(esConsumes<TrackerTopology, TrackerTopologyRcd>()),
```
Comment (Dr15Jones) [inline CalibTracker/SiStripHitEfficiency/src/HitEff.cc:82]: For future reference, one can now just write
```C++
topoToken_(esConsumes());
```

Comment (mmusich) [reply CalibTracker/SiStripHitEfficiency/src/HitEff.cc]: thanks @Dr15Jones, much more compact indeed. 
Applied here, wherever relevant.

**Comment by mmusich** [comment]
@cmsbuild, please test

File: CalibTracker/SiStripHitEfficiency/src/HitEff.cc [hunk 2]
```diff
@@ -159,8 +165,7 @@ void HitEff::beginJob() {
 
 void HitEff::analyze(const edm::Event& e, const edm::EventSetup& es) {
   //Retrieve tracker topology from geometry
-  edm::ESHandle<TrackerTopology> tTopoHandle;
-  es.get<TrackerTopologyRcd>().get(tTopoHandle);
+  edm::ESHandle<TrackerTopology> tTopoHandle = es.getHandle(topoToken_);
   const TrackerTopology* const tTopo = tTopoHandle.product();
```
Comment (Dr15Jones) [inline CalibTracker/SiStripHitEfficiency/src/HitEff.cc:169]: This can be simplified to 
```
const TrackerTopology& tTopo = es.getData(topoToken_);
```

Comment (mmusich) [reply CalibTracker/SiStripHitEfficiency/src/HitEff.cc]: thanks. Simplified now.

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by yuanchao** [comment]
+1

**Comment by silviodonato** [comment]
+1

**Pull request merged**